config/dbstructure.json

```diff
@@ -15,6 +15,34 @@
 			"name": ["UNIQUE", "name"]
 		}
 	},
+	"apcontact": {
```

Comment:
I don't like having a separate `contact` table for a specific protocol, can't we include the relevant fields in the current `contact` table?

Reply:
No. For some time we do have contacts that are supporting multiple protocols. This table is the sibling to the existing "fcontact" table that does the same - but for the diaspora protocol. There are several fields that only AP supports or that have a very different notation. And it's a waste of space duplicating all this data in a "per user" contact table.

Reply:
I understand. I'd really like us to have a unified contact table with all the common fields for any protocol, so that we can have a simple way of storing the relationship between users and contacts. Can we have the protocol-specific data in separate tables instead?

Reply:
This table already only does hold the protocol specific data.

Reply:
What about the `url`, `nick`, `name`, `photo`, `addr` and `alias` fields?

Reply:
All this data is fetched via the probing, so I would like to keep it together. This table is very encapsulated.

Reply:
That's exactly my problem! I'd like it to be interacting more with the main `contact` table to retrieve/update common fields instead of duplicating them. I understand why you would have a separate table during development, but ultimately I'd like the common fields to only belong in the `contact` table.

Reply:
I would like to separate all the technical stuff out of the contact table.

Reply:
So it would be great, having all data from the different networks in different tables. Then we would only copy the needed stuff to the contact table.

Reply:
I wouldn't want to make drastic changes in this case to this PR.

Reply:
Fair enough, as long as it is in the back of your mind.

Reply:
I'm still unsure, how to do this all, but normally I'm getting good ideas while not thinking about them.